### PR TITLE
PN-350: Add ability to filter out cases where owned match is solved

### PR DESCRIPTION
**Pull request by veronikaslc**
Done.

### CODE REVIEW

File: matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml [hunk 1]
```diff
@@ -161,6 +161,9 @@ $xwiki.jsx.use('PhenoTips.Widgets')##
                 &lt;label&gt;&lt;input type="checkbox" name="access-filter" value="owner" checked /&gt;$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.owner'))&lt;/label&gt;&lt;br&gt;
                 &lt;label&gt;&lt;input type="checkbox" name="access-filter" value="view" checked /&gt;$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.view'))&lt;/label&gt;
               &lt;/li&gt;
+              &lt;li&gt;$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.solvedFilter'))&lt;br&gt;
+                &lt;label&gt;&lt;input type="checkbox" name="solved-filter" value="hide"/&gt;$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.solved'))&lt;/label&gt;&lt;br&gt;
```
Comment (sdumitriu) [inline matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml]: Is the `br` at the end necessary?

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js [hunk 2]
```diff
@@ -372,8 +380,20 @@ var PhenoTips = (function (PhenoTips) {
...
+            var matchHasOwnSolvedCase = function(match) {
+                    if (match.reference.serverId == "" && match.reference.ownership["userIsOwner"] && match.reference.pubmedIDs && match.reference.pubmedIDs.size() > 0) {
+                        return true;
+                    }
+                    if (match.matched.serverId == "" && match.matched.ownership["userIsOwner"] && match.matched.pubmedIDs && match.matched.pubmedIDs.size() > 0) {
+                        return true;
+                    }
+                    return false;
+                };
+            var hideOwnSolvedSaces = !this._filterValues.solved || !matchHasOwnSolvedCase;
```
Comment (sdumitriu) [inline matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: Typo: `saces`?

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js [hunk 3]
```diff
@@ -372,8 +380,20 @@ var PhenoTips = (function (PhenoTips) {
             var isNotifiedMatch = match.notified && this._filterValues.notified.notified || !match.notified && this._filterValues.notified.unnotified;
             var hasScoreMatch = match.score >= this._filterValues.score.score
                              && match.phenotypicScore >= this._filterValues.score.phenotypicScore
-                             && match.genotypicScore >= this._filterValues.score.genotypicScore;
-            return hasExternalIdMatch && hasEmailMatch && hasGeneSymbolMatch && hasAccessTypeMath && hasOwnershipMatch
+
+            // returns true if one of the records in match is local and owned my user and is solved
+            var matchHasOwnSolvedCase = function(match) {
+                    if (match.reference.serverId == "" && match.reference.ownership["userIsOwner"] && match.reference.pubmedIDs && match.reference.pubmedIDs.size() > 0) {
```
Comment (sdumitriu) [inline matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: I feel that the check shouldn't be on `pubmedIds.size > 0` (which is done in too many places), but on a simple `solved` boolean.

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js (hunk 2 above)
Comment (veronikaslc) [reply matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: yup

File: matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml (hunk 1 above)
Comment (veronikaslc) [reply matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml]: will remove

**Comment by veronikaslc** [comment]
Addressed codereview comments.

**Pull request merged**